[BUG] Flaky test: TestIntegration_GracefulShutdownWithPendingEvents

## Description

The test `TestIntegration_GracefulShutdownWithPendingEvents` in `internal/ingestion/integration_test.go` is flaky and intermittently fails in CI.

## Failure Message

```
--- FAIL: TestIntegration_GracefulShutdownWithPendingEvents (0.50s)
    integration_test.go:437: expected some events to be processed during drain
    integration_test.go:439: processed 0 events during graceful shutdown
```

## Root Cause

The test expects events to be processed during graceful shutdown drain, but due to timing issues in CI (slower machines, resource contention), sometimes no events are processed before the shutdown completes.

## Impact

- CI fails intermittently
- Blocks PRs from being merged
- Requires manual re-runs

## Observed Frequency

Seen multiple times in CI runs. The test passed on rerun in some cases.

## Suggested Fix

Options:
1. Increase the drain timeout in the test
2. Add explicit synchronization to ensure events are queued before shutdown
3. Make the test more deterministic by pre-populating the event channel
4. Mark the test as flaky and skip in CI (not recommended)

## Related CI Runs

- https://github.com/Obsyk/obsyk-operator/actions/runs/20053739049/job/57514974358

--- a/Makefile
+++ b/Makefile
@@ -17,7 +17,7 @@ GIT_VERSION ?= $(shell git describe --tags --always --dirty 2>/dev/null || echo 
 LDFLAGS := -X main.version=$(GIT_VERSION)
 
 # Tool versions
-CONTROLLER_GEN_VERSION ?= v0.14.0
+CONTROLLER_GEN_VERSION ?= v0.19.0
 GOLANGCI_LINT_VERSION ?= v1.62.2
 
 # Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
@@ -84,11 +84,12 @@ generate: controller-gen ## Generate code (DeepCopy methods)
 .PHONY: manifests
 manifests: controller-gen ## Generate CRD manifests
 	$(CONTROLLER_GEN) crd paths="./api/..." output:crd:artifacts:config=config/crd/bases
+	@cp config/crd/bases/*.yaml charts/obsyk-operator/crds/
 
 ##@ CI/CD
 
 .PHONY: ci
-ci: lint test build ## Run all CI checks (lint, test, build)
+ci: lint test build verify-manifests ## Run all CI checks (lint, test, build, verify-manifests)
 
 .PHONY: verify
 verify: fmt ## Verify code formatting (for CI)
@@ -97,6 +98,15 @@ verify: fmt ## Verify code formatting (for CI)
 		git diff; \
 		exit 1; \
 	fi
+
+.PHONY: verify-manifests
+verify-manifests: manifests ## Verify CRD manifests are up to date
+	@if [ -n "$$(git diff --name-only config/crd/ charts/obsyk-operator/crds/)" ]; then \
+		echo "CRD manifests are out of date. Run 'make manifests' and copy to charts/obsyk-operator/crds/"; \
+		git diff config/crd/ charts/obsyk-operator/crds/; \
+		exit 1; \
+	fi
+	@echo "CRD manifests are up to date"
 
 ##@ Tools
 

--- a/charts/obsyk-operator/crds/obsyk.io_obsykagents.yaml
+++ b/charts/obsyk-operator/crds/obsyk.io_obsykagents.yaml
@@ -3,7 +3,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.14.0
+    controller-gen.kubebuilder.io/version: v0.19.0
   name: obsykagents.obsyk.io
 spec:
   group: obsyk.io
@@ -132,16 +132,8 @@ spec:
                 description: Conditions represent the latest available observations
                   of the agent's state.
                 items:
-                  description: "Condition contains details for one aspect of the current
-                    state of this API Resource.\n---\nThis struct is intended for
-                    direct use as an array at the field path .status.conditions.  For
-                    example,\n\n\n\ttype FooStatus struct{\n\t    // Represents the
-                    observations of a foo's current state.\n\t    // Known .status.conditions.type
-                    are: \"Available\", \"Progressing\", and \"Degraded\"\n\t    //
-                    +patchMergeKey=type\n\t    // +patchStrategy=merge\n\t    // +listType=map\n\t
-                    \   // +listMapKey=type\n\t    Conditions []metav1.Condition `json:\"conditions,omitempty\"
-                    patchStrategy:\"merge\" patchMergeKey:\"type\" protobuf:\"bytes,1,rep,name=conditions\"`\n\n\n\t
-                    \   // other fields\n\t}"
+                  description: Condition contains details for one aspect of the current
+                    state of this API Resource.
                   properties:
                     lastTransitionTime:
                       description: |-
@@ -182,12 +174,7 @@ spec:
                       - Unknown
                       type: string
                     type:
-                      description: |-
-                        type of condition in CamelCase or in foo.example.com/CamelCase.
-                        ---
-                        Many .condition.type values are consistent across resources like Available, but because arbitrary conditions can be
-                        useful (see .node.status.conditions), the ability to deconflict is important.
-                        The regex it matches is (dns1123SubdomainFmt/)?(qualifiedNameFmt)
+                      description: type of condition in CamelCase or in foo.example.com/CamelCase.
                       maxLength: 316
                       pattern: ^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*/)?(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])$
                       type: string
@@ -220,16 +207,90 @@ spec:
               resourceCounts:
                 description: ResourceCounts tracks the number of watched resources.
                 properties:
+                  clusterRoleBindings:
+                    description: ClusterRoleBindings is the count of cluster role
+                      bindings being watched.
+                    format: int32
+                    type: integer
+                  clusterRoles:
+                    description: ClusterRoles is the count of cluster roles being
+                      watched.
+                    format: int32
+                    type: integer
+                  configMaps:
+                    description: ConfigMaps is the count of configmaps being watched.
+                    format: int32
+                    type: integer
+                  cronJobs:
+                    description: CronJobs is the count of cronjobs being watched.
+                    format: int32
+                    type: integer
+                  daemonSets:
+                    description: DaemonSets is the count of daemonsets being watched.
+                    format: int32
+                    type: integer
+                  deployments:
+                    description: Deployments is the count of deployments being watched.
+                    format: int32
+                    type: integer
+                  events:
+                    description: Events is the count of events being watched.
+                    format: int32
+                    type: integer
+                  ingresses:
+                    description: Ingresses is the count of ingresses being watched.
+                    format: int32
+                    type: integer
+                  jobs:
+                    description: Jobs is the count of jobs being watched.
+                    format: int32
+                    type: integer
                   namespaces:
                     description: Namespaces is the count of namespaces being watched.
+                    format: int32
+                    type: integer
+                  networkPolicies:
+                    description: NetworkPolicies is the count of network policies
+                      being watched.
+                    format: int32
+                    type: integer
+                  nodes:
+                    description: Nodes is the count of nodes being watched.
                     format: int32
                     type: integer
                   pods:
                     description: Pods is the count of pods being watched.
                     format: int32
                     type: integer
+                  pvcs:
+                    description: PVCs is the count of persistent volume claims being
+                      watched.
+                    format: int32
+                    type: integer
+                  roleBindings:
+                    description: RoleBindings is the count of role bindings being
+                      watched.
+                    format: int32
+                    type: integer
+                  roles:
+                    description: Roles is the count of roles being watched.
+                    format: int32
+                    type: integer
+                  secrets:
+                    description: Secrets is the count of secrets being watched.
+                    format: int32
+                    type: integer
+                  serviceAccounts:
+                    description: ServiceAccounts is the count of service accounts
+                      being watched.
+                    format: int32
+                    type: integer
                   services:
                     description: Services is the count of services being watched.
+                    format: int32
+                    type: integer
+                  statefulSets:
+                    description: StatefulSets is the count of statefulsets being watched.
                     format: int32
                     type: integer
                 type: object

--- a/config/crd/bases/obsyk.io_obsykagents.yaml
+++ b/config/crd/bases/obsyk.io_obsykagents.yaml
@@ -3,7 +3,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.14.0
+    controller-gen.kubebuilder.io/version: v0.19.0
   name: obsykagents.obsyk.io
 spec:
   group: obsyk.io
@@ -132,16 +132,8 @@ spec:
                 description: Conditions represent the latest available observations
                   of the agent's state.
                 items:
-                  description: "Condition contains details for one aspect of the current
-                    state of this API Resource.\n---\nThis struct is intended for
-                    direct use as an array at the field path .status.conditions.  For
-                    example,\n\n\n\ttype FooStatus struct{\n\t    // Represents the
-                    observations of a foo's current state.\n\t    // Known .status.conditions.type
-                    are: \"Available\", \"Progressing\", and \"Degraded\"\n\t    //
-                    +patchMergeKey=type\n\t    // +patchStrategy=merge\n\t    // +listType=map\n\t
-                    \   // +listMapKey=type\n\t    Conditions []metav1.Condition `json:\"conditions,omitempty\"
-                    patchStrategy:\"merge\" patchMergeKey:\"type\" protobuf:\"bytes,1,rep,name=conditions\"`\n\n\n\t
-                    \   // other fields\n\t}"
+                  description: Condition contains details for one aspect of the current
+                    state of this API Resource.
                   properties:
                     lastTransitionTime:
                       description: |-
@@ -182,12 +174,7 @@ spec:
                       - Unknown
                       type: string
                     type:
-                      description: |-
-                        type of condition in CamelCase or in foo.example.com/CamelCase.
-                        ---
-                        Many .condition.type values are consistent across resources like Available, but because arbitrary conditions can be
-                        useful (see .node.status.conditions), the ability to deconflict is important.
-                        The regex it matches is (dns1123SubdomainFmt/)?(qualifiedNameFmt)
+                      description: type of condition in CamelCase or in foo.example.com/CamelCase.
                       maxLength: 316
                       pattern: ^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*/)?(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])$
                       type: string
@@ -220,16 +207,90 @@ spec:
               resourceCounts:
                 description: ResourceCounts tracks the number of watched resources.
                 properties:
+                  clusterRoleBindings:
+                    description: ClusterRoleBindings is the count of cluster role
+                      bindings being watched.
+                    format: int32
+                    type: integer
+                  clusterRoles:
+                    description: ClusterRoles is the count of cluster roles being
+                      watched.
+                    format: int32
+                    type: integer
+                  configMaps:
+                    description: ConfigMaps is the count of configmaps being watched.
+                    format: int32
+                    type: integer
+                  cronJobs:
+                    description: CronJobs is the count of cronjobs being watched.
+                    format: int32
+                    type: integer
+                  daemonSets:
+                    description: DaemonSets is the count of daemonsets being watched.
+                    format: int32
+                    type: integer
+                  deployments:
+                    description: Deployments is the count of deployments being watched.
+                    format: int32
+                    type: integer
+                  events:
+                    description: Events is the count of events being watched.
+                    format: int32
+                    type: integer
+                  ingresses:
+                    description: Ingresses is the count of ingresses being watched.
+                    format: int32
+                    type: integer
+                  jobs:
+                    description: Jobs is the count of jobs being watched.
+                    format: int32
+                    type: integer
                   namespaces:
                     description: Namespaces is the count of namespaces being watched.
+                    format: int32
+                    type: integer
+                  networkPolicies:
+                    description: NetworkPolicies is the count of network policies
+                      being watched.
+                    format: int32
+                    type: integer
+                  nodes:
+                    description: Nodes is the count of nodes being watched.
                     format: int32
                     type: integer
                   pods:
                     description: Pods is the count of pods being watched.
                     format: int32
                     type: integer
+                  pvcs:
+                    description: PVCs is the count of persistent volume claims being
+                      watched.
+                    format: int32
+                    type: integer
+                  roleBindings:
+                    description: RoleBindings is the count of role bindings being
+                      watched.
+                    format: int32
+                    type: integer
+                  roles:
+                    description: Roles is the count of roles being watched.
+                    format: int32
+                    type: integer
+                  secrets:
+                    description: Secrets is the count of secrets being watched.
+                    format: int32
+                    type: integer
+                  serviceAccounts:
+                    description: ServiceAccounts is the count of service accounts
+                      being watched.
+                    format: int32
+                    type: integer
                   services:
                     description: Services is the count of services being watched.
+                    format: int32
+                    type: integer
+                  statefulSets:
+                    description: StatefulSets is the count of statefulsets being watched.
                     format: int32
                     type: integer
                 type: object

--- a/internal/ingestion/integration_test.go
+++ b/internal/ingestion/integration_test.go
@@ -419,7 +419,11 @@ func TestIntegration_GracefulShutdownWithPendingEvents(t *testing.T) {
 		_, _ = clientset.CoreV1().Pods("default").Create(ctx, pod, metav1.CreateOptions{})
 	}
 
-	// Stop manager immediately - should drain remaining events
+	// Give informer time to queue events before stopping
+	// This prevents flaky failures in CI where events aren't queued yet
+	time.Sleep(100 * time.Millisecond)
+
+	// Stop manager - should drain remaining events
 	mgr.Stop()
 
 	select {

